src/pages/MapViewer/components/Toolbox/components/item-assets/AssetsPanel.tsx

```diff
@@ -0,0 +1,49 @@
+import { ClipboardButton } from '@/components/clipboard/ClipboardButton';
+import { useToolbox } from '@/hooks/useToolbox';
+
+import './AssetsPanel.scss';
+
+export const AssetsPanel = () => {
+  const {
+    state: { selectedCollectionItem },
+    actions: { setActivePage },
+  } = useToolbox();
+
+  // Reduce the object of objects down to an array of key/value objects.
+  const assets = Object.keys(selectedCollectionItem.assets).reduce((acc, key) => {
```

Comment:
This can be typed: https://github.com/EO-DataHub/eodhp-resource-catalogue-ui/blob/main/src/typings/common.ts#L30

Reply:
Not really, or at least not as you suggest. The `assets` variable is not a `StacAsset[]`, but a simple array of objects of `key/href` e.g.

```JavaScript
[
  { cloud: 'http://....' },
  ...
  { valid_pixels: 'http://....' },
]
```
I am find with creating a local `type`:

```JavaScript
type Asset = {
  key: string;
  value: string;
}
```
That would satisfy typing of the reduced array, but since the reduce is using only the `keys` from the assets and not the assets from the CollectionItem themselves, what you suggest wouldn't work.

the StacAsset is an object and I only care about the key name and the href value, so I tend to find it easier to reduce this object down to a simple array of new objects, in the shape that is easy to iterate, to create the label/input/button structure. I could not do this and instead of the `assets.map` later, do a more complicated extraction of the keys and href value there. The code would be very similar but I feel that makes the code less readable.

Reply:
Okay good point